src/core/server/csp/config.ts

```diff
@@ -7,28 +7,56 @@
  */
 
 import { TypeOf, schema } from '@kbn/config-schema';
+import { ServiceConfigDescriptor } from '../internal_types';
+
+const configSchema = schema.object(
+  {
+    rules: schema.maybe(schema.arrayOf(schema.string())),
+    script_src: schema.arrayOf(schema.string(), { defaultValue: [] }),
+    worker_src: schema.arrayOf(schema.string(), { defaultValue: [] }),
+    style_src: schema.arrayOf(schema.string(), { defaultValue: [] }),
```

Comment:
Is there any particular reason you chose not to use hyphens for these keys? E.g.,

```ts
    'script-src': schema.arrayOf(schema.string(), { defaultValue: [] }),
    'worker-src': schema.arrayOf(schema.string(), { defaultValue: [] }),
    'style-src': schema.arrayOf(schema.string(), { defaultValue: [] }),
```

Reply:
1. dev experience, basically, to avoid accessing the props via `config['script-src']` when manipulating it.
2. configuration key naming convention (which is a more valid reason).

Reply:
RE: convention, we mostly use camelCase, which is part of why it stuck out to me. But I guess we do have a few config keys with underscores such as `xpack.license_management.enabled`. I think my preference is still hyphens because we are referencing a standard that uses hyphens, but this isn't the hill that I'm willing to die on 😄

Reply:
> RE: convention, we mostly use camelCase, which is part of why it stuck out to me

The long-term plan is to align with the stack and only use snake_case naming for config keys. See https://github.com/elastic/kibana/issues/7444. `core` is trying to only use snake_case for new keys to reduce the cost of this conversion. 

@joshdover WDTY? Do you think it's better to go with the spec's names and use `script-src` instead of `script_src`?

Reply:
I'd rather align with the snake_case config key convention over the CSP spec. Each of these should be explicitly documented anyways so there's clarity on which options are available.